### PR TITLE
Add metrics for accounts and storage reads (Flat databasae vs MPT)

### PR DESCRIPTION
**Pull request by ahamlat**
## PR description
The objective of this PR is to introduce 8 new metrics (counters) that will track important aspects of account and storage reads. These metrics will include:

- Total number of calls (for both accounts and storage)
- Number of account or storage entries found in the flat database
- Number of storage slots located in the merkle trie but not found in the flat database
- Number of account or storage entries not found in either the flat database or the merkle trie

The insights provided by these new metrics will be visualized through Grafana graphs, which will enable us to better monitor and analyze the performance of accounts and storage reads. By leveraging these graphs, we can gain a deeper understanding of the state of the flat database, the number of missing entries in Merkle Patricia Trie database and take steps to optimize performance.

It is important to keep in mind that we're not counting here all the calls to the Patricia Merkle Trie for accounts and storage, but we're are counting only 1 for getting the leaf from the trie, which in practice corresponds to several calls.

image

### CODE REVIEW

**Review by matkt — APPROVED** [review]
Regarding the metrics LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### Additions and Improvements
 - Update most dependencies to latest version [#5269](https://github.com/hyperledger/besu/pull/5269)
+- Add metrics for accounts and storage reads (Flat databasae vs Merkle Patricia Trie) [#5315](https://github.com/hyperledger/besu/pull/5315)
```
Comment (macfarla) [inline CHANGELOG.md:7]:
```suggestion
- Add metrics for accounts and storage reads (Flat database vs Merkle Patricia Trie) [#5315](https://github.com/hyperledger/besu/pull/5315)
```

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/storage/BonsaiWorldStateKeyValueStorage.java [hunk 2]
```diff
@@ -55,32 +58,96 @@ public class BonsaiWorldStateKeyValueStorage implements WorldStateStorage, AutoC
...
+        metricsSystem.createCounter(
+            BesuMetricCategory.BLOCKCHAIN,
+            "get_account_merkle_trie",
+            "Number of accounts not found in the flat database, but found in the merkle trie");
+
+    getAccountMissingMerkleTrieCounter =
+        metricsSystem.createCounter(
+            BesuMetricCategory.BLOCKCHAIN,
+            "get_account_missing_merkle_trie",
+            "Number of accounts not found in the flat database and in the merkle trie");
```
Comment (macfarla) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/storage/BonsaiWorldStateKeyValueStorage.java:126]:
```suggestion
            "Number of accounts not found (either in the flat database or the merkle trie)");
```

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/storage/BonsaiWorldStateKeyValueStorage.java [hunk 3]
```diff
@@ -55,32 +58,96 @@ public class BonsaiWorldStateKeyValueStorage implements WorldStateStorage, AutoC
...
+        metricsSystem.createCounter(
+            BesuMetricCategory.BLOCKCHAIN,
+            "get_storagevalue_merkle_trie",
+            "Number of storage slots not found in the flat database, but found in the merkle trie");
+
+    getStorageValueMissingMerkleTrieCounter =
+        metricsSystem.createCounter(
+            BesuMetricCategory.BLOCKCHAIN,
+            "get_storagevalue_missing_merkle_trie",
+            "Number of storage slots not found in the flat database and in the merkle trie");
```
Comment (macfarla) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/storage/BonsaiWorldStateKeyValueStorage.java:150]:
```suggestion
            "Number of storage slots not found (either in the flat database or in the merkle trie)");
```

**Review by macfarla — APPROVED** [review]
couple of suggestions on the metric names

**Pull request merged**